### PR TITLE
Adding possibility to change control modes in history matching

### LINKED ISSUE #219: Enable HM controls in the FlowNet config file
It could be useful to enable HM controls / targets for producers (WCONHIST) and injectors (WCONINJH) to be set in the FlowNet config file. Right now, the target 'RESV' is set for producers (where a few other options would be possible) and 'RATE' for injectors (where it could be also 'BHP').

### PR DESCRIPTION
**Pull request by olelod**
Adding the possibility to change how the production wells are controlled in the WCONHIST keywords. This will not be controlled by one keyword in the config yaml.

---

### Contributor checklist

- [x] :tada: This PR closes #219 
- [x] :scroll: I have broken down my PR into the following tasks:
   - [x] Setting keyword for control mode in WCONHIST and WCONINJH in the config yaml
- [ ] :robot: I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR.
- [x] :book: I have considered adding a new entry in `CHANGELOG.md`.
- [ ] :books: I have considered updating the documentation.

### CODE REVIEW

File: src/flownet/realization/_schedule.py [hunk 1]
```diff
@@ -197,6 +197,7 @@ def _calculate_wconinjh(self):
                         rate=value["WWIR"],
                         bhp=value["WBHP"],
                         thp=value["WTHP"],
+                        target=self._inj_control_mode,
```
Comment (wouterjdb) [inline src/flownet/realization/_schedule.py:200]: And then this `inj_control_mode`?

File: src/flownet/realization/_schedule.py [hunk 2]
```diff
@@ -494,35 +496,30 @@ def get_nr_observations(self, training_set_fraction: float) -> int:
         num_training_dates = round(len(self.get_dates()) * training_set_fraction)
         i = 0
 
-        for date in self.get_dates()[0 : num_training_dates - 1]:
+        for date in self.get_dates()[1 : num_training_dates - 1]:
```
Comment (wouterjdb) [inline src/flownet/realization/_schedule.py:499]: Why should this be 1? Now you are skipping the first date?

**Review by wouterjdb — CHANGES_REQUESTED** [review]

File: src/flownet/realization/_schedule.py (hunk 2 above)
Comment (olelod) [reply src/flownet/realization/_schedule.py:499]: For some reason the code used to remove the first date for all wells. I removed that, but then ERT gives an error about the first date of the simulation: `** ERROR: It is unfortunately not possible to use summary observations from the
          start of the simulation. Problem with observation:WGPR_B-2H_06_11_1997 at 06/11/1997`. Then i just removed the first date (both in generating the `observations.ertobs` and in counting the number of observations).

File: src/flownet/realization/_schedule.py [hunk 3]
```diff
@@ -162,7 +162,7 @@ def _calculate_wconhist(self):
                         date=value["date"],
                         well_name=value["WELL_NAME"],
                         status=value["WSTAT"],
-                        control_mode=self._control_mode,
+                        control_mode=self._prod_control_mode,
```
[the remark this answers is not in the document]
Comment (olelod) [reply src/flownet/realization/_schedule.py:165]: I just used the ones allready defined, but i agree with your comment about clarity.

File: src/flownet/realization/_schedule.py (hunk 1 above)
Comment (olelod) [reply src/flownet/realization/_schedule.py:200]: Same here

**Review by wouterjdb — APPROVED** [review]
LGTM 🚀

**Pull request merged**